inference/utils/inference_api_client.py

```diff
@@ -0,0 +1,185 @@
+            return {"status": JobStatusEnum.FAILED, "message": f"API request failed: {str(e)}"}
+
+    def load_model(self, model_identifier: str) -> bool:
+        """Request model loading from API"""
+        response = self.make_api_request("POST", f"{model_identifier}/load")
+        return response is not None and response.get("status") != JobStatusEnum.FAILED
+
+    def unload_model(self, model_identifier: str) -> bool:
+        """Request model unloading from API"""
+        response = self.make_api_request("POST", f"{model_identifier}/unload")
+        return response is not None and response.get("status") != JobStatusEnum.FAILED
+
+    def check_model_status(self, model_identifier: str) -> ModelStatusEnum:
+        """Check if model is loaded and ready"""
+        response = self.make_api_request("GET", f"{model_identifier}/status")
+        if not response:
+            return ModelStatusEnum.UNKNOWN
+
+        try:
+            return ModelStatusEnum(response.get("status", ModelStatusEnum.UNKNOWN))
+        except ValueError:
+            return ModelStatusEnum.UNKNOWN
+
+    def submit_batch(self, model_identifier: str, batch_data: list[BatchItem]) -> str | None:
+        """Submit a batch of items for inference"""
+        if not batch_data:
+            return None
+
+        # Validate and extract text data
+        try:
+            text_data = [item["text"] for item in batch_data]
+        except (KeyError, TypeError):
+            return None
+
+        response = self.make_api_request("POST", f"{model_identifier}/jobs", json={"input_data": text_data})
+        return response.get("job_id") if response else None
+
+    def get_job_status(self, model_identifier: str, job_id: str) -> JobResponse:
+        """Check status of a submitted job"""
+        response = self.make_api_request("GET", f"{model_identifier}/jobs/{job_id}")
+        if not response:
+            return {
+                "status": JobStatusEnum.FAILED,
+                "message": "Failed to get job status",
+                "job_id": job_id,
+                "results": None,
+            }
+        return response
+
+    def get_available_inferencers(self) -> dict:
+        """Get all available inferencer models"""
+        response = self.make_api_request("GET", "")
+        return response if response else {}
+
+
+class ModelManager:
+    """Handles model loading/unloading and ensures only the specified model is loaded"""
+
+    def __init__(self, api_client: InferenceAPIClient, model_identifier: str):
+        """Initialize with API client and the model to manage
+
+        Args:
+            api_client: InferenceAPIClient instance
+            model_identifier: The model this manager will handle
+        """
+        self.api_client = api_client
+        self.model_identifier = model_identifier
+
+    def _unload_all_other_models(self) -> bool:
+        """Unload all models except the one managed by this instance"""
```

Comment:
Need to implement 
_request_model_unload: sends the unload request
unload_model: calls request, then send status check request until confirmation of unload, with error handling if max retries met
unload_all_models:run _unload for all models

then we can implement a similar pattern on the loading side as well. 
Lastly, lets merge model manager into apiclient, since there is no logical separation.